### PR TITLE
Update documents with general information on Kyma

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Updated the main README.md document
- Updated the general Overview document for Kyma
- Added additional Overview documents
- Changed the ordering of the Details documents as they were mixed up

**Related issue(s)**
See also https://github.com/kyma-project/kyma/issues/611

### CODE REVIEW

**Comment by kazydek** [comment]
@pbochynski @PK85 @lszymik @joek Hi Guys, pls take a look at this PR - this is a follow-up task on the issue re the landing page content that we were discussing some time ago: https://github.com/kyma-project/kyma/issues/611. A sort of GS guide was created as a result. We will introduce it to Kyma in separate Overview docs. Some of that content will be also used on the landing page - see the related issue for that task: https://github.com/kyma-project/website/issues/45. Let me know if you have any comemnts. Thanks!

File: README.md [hunk 1]
```diff
@@ -65,7 +53,7 @@ Every project runs differently. Follow instructions in the main `README.md` docu
 Do not forget about creating unit and acceptance tests if needed. For the unit tests, follow the instructions specified in the main `README.md` document of the given project. For the details concerning the acceptance tests, go to the corresponding directory inside the `tests` directory.
 Find the information on how to run changes on the cluster without a Docker image in the [Develop a service locally without using Docker](docs/kyma/docs/035-gs-local-develop-no-docker.md) document.
 
->**NOTE:** For more details about testing, go to the [Testing Kyma](docs/kyma/docs/026-details-testing.md) document.
+>**NOTE:** For more details about testing, go to the [Testing Kyma](docs/kyma/docs/013-details-testing.md) document.
```
Comment (derberg) [inline README.md]: link here https://kyma-project.io/docs/latest/root/kyma#details-testing-kyma
we should not promote reading from github

File: resources/core/charts/environments/README.md [hunk 2]
```diff
@@ -11,4 +11,4 @@
 
 Environments is the name of the controller that injects limit ranges and resource quotas into user-created Namespaces. It also adds two default roles, `kyma-reader-role` and `kyma-admin-role`, which it then copies into user-created Environments.
 
-Read the [environments](../../../../docs/kyma/docs/005-details-environments.md) document to learn more about Environments in Kyma.
+Read the [environments](../../../../docs/kyma/docs/011-details-environments.md) document to learn more about Environments in Kyma.
```
Comment (derberg) [inline resources/core/charts/environments/README.md]: link to kyma-project please

File: docs/kyma/docs/005-overview-kyma.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+title: Kyma and Knative - brothers in arms
```
Comment (derberg) [inline docs/kyma/docs/005-overview-kyma.md]: please extend doc with https://github.com/kyma-project/website/pull/49/files just one paragraph, that nearest plans are not only Istion exchange but also replacement of Kubeless

File: README.md (hunk 1 above)
Comment (kazydek) [reply README.md]: Ok. In that case, I also changed links for the local and cluster installation. I guess we should talk on the Content SIG on that approach as well as I can expect no one knows that is the new approach.

File: docs/kyma/docs/005-overview-kyma.md (hunk 3 above)
Comment (kazydek) [reply docs/kyma/docs/005-overview-kyma.md]: Will do.

File: docs/kyma/docs/005-overview-kyma.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+The diagram shows dependencies between the components:
+
+![kyma-knative](./assets/kyma-knative.svg)
+
+### Planned changes
+
+The nearest plan for Kyma and Knative cooperation is to replace Kubeless with the Knative technology. The plan assumes to take the Knative `Serving` and `Build` components as the building blocks for the new architecture, and to introduce new custom components to bridge any existing gaps.
+
+These new components include:
+- a custom build template that provides the function interface available in Kubeless
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/kyma/docs/005-overview-kyma.md]: Sure, I will check that with @joek.

File: docs/kyma/docs/005-overview-kyma.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+- a custom docker registry to store the build artifacts
+- a storage solution to store the function code (a Git repository or a blob storage like Minio or S3)
+
+The implementation process involves multiple components and and will follow these steps:
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/kyma/docs/005-overview-kyma.md]: Will consult it with @joek as well.

File: README.md [hunk 6]
```diff
@@ -31,7 +19,7 @@ Kyma comes with the ready-to-use code snippets that you can use to test the exte
...
 
 Develop on your remote repository forked from the original repository in Go.
-See the example that uses the [`ui-api-layer`](components/ui-api-layer) project located in the `components` directory in the `kyma` repository but applies to any Go project. This set of instructions uses the recommended [`git workflow`](https://github.com/kyma-project/community/blob/master/git-workflow.md) and the general [contribution flow](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#contribute-code-or-content). Read also the [`CONTRIBUTING.md`] document that includes the contributing rules specific for this repository.
+See the example that uses the [`ui-api-layer`](./components/ui-api-layer) project located in the `components` directory in the `kyma` repository but applies to any Go project. This set of instructions uses the recommended [`git workflow`](https://github.com/kyma-project/community/blob/master/git-workflow.md) and the general [contribution flow](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#contribute-code-or-content). Read also the [`CONTRIBUTING.md`] document that includes the contributing rules specific for this repository.
```
Comment (klaudiagrz) [inline README.md]: 1. "See the example that uses the [`ui-api-layer`](./components/ui-api-layer) project located in the `components` directory in the `kyma` repository but applies to any Go project." -> ``See the example that uses the [`ui-api-layer`](./components/ui-api-layer) project located under the `kyma/components` directory but applies to any Go project.``
2. [`CONTRIBUTING.md`] document -> there's a link missing in here

File: docs/kyma/docs/004-overview-kyma.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+Kyma is built of numerous components but these three drive it forward:
+
+  - **Application Connector** connects external on-premise or cloud-native applications with Kyma and registers their APIs and Events that internal applications or functions can later consume.
+  - **Serverless** is based on [Kubeless](https://kubeless.io/) and allows you to write short code snippets, known as functions or lambdas, that consume the Events exposed by the Application Connector and use external services provided by the Service Catalog to trigger certain business logic.
+  - **Service Catalog** offers third-party services in the form of ServiceClasses exposed by the Service Brokers that a given function can consume to perform certain actions.
+
+This basic use case shows how the three components work together in Kyma:
+
+![key-components](./assets/ac-s-sc.svg)
```
Comment (klaudiagrz) [inline docs/kyma/docs/004-overview-kyma.md]: On the diagram: `Expose services through the Service Brokers`

File: docs/kyma/docs/005-overview-kyma.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+![kyma-knative](./assets/kyma-knative.svg)
+
+### Planned changes
+
+The nearest plan for Kyma and Knative cooperation is to replace Kubeless with the Knative technology. The plan assumes to take the Knative `Serving` and `Build` components as the building blocks for the new architecture, and to introduce new custom components to bridge any existing gaps.
+
+These new components include:
+- a custom build template that provides the function interface available in Kubeless
+- a custom docker registry to store the build artifacts
```
Comment (klaudiagrz) [inline docs/kyma/docs/005-overview-kyma.md]: `Docker`

File: docs/kyma/docs/005-overview-kyma.md (hunk 5 above)
Comment (klaudiagrz) [inline docs/kyma/docs/005-overview-kyma.md]: ...multiple components ~and~ and will...

File: docs/kyma/docs/005-overview-kyma.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+- a storage solution to store the function code (a Git repository or a blob storage like Minio or S3)
+
+The implementation process involves multiple components and and will follow these steps:
+1. Integration of Knative `Serving` and `Build` components as optional Kyma modules.
+2. Enabling other parts like docker registry, blob storage, and build template (the order is not defined yet).
```
Comment (klaudiagrz) [inline docs/kyma/docs/005-overview-kyma.md]: `Docker`

File: README.md (hunk 6 above)
Comment (kazydek) [reply README.md]: Thanks for pointing out the link. I can shorten the path as well. As for the use of "in" or "under" I guess it is rather a matter of preference.

File: docs/kyma/docs/005-overview-kyma.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (joek) [reply docs/kyma/docs/005-overview-kyma.md]: Actually all the planned changes are not fully sketched out. I am not sure if the concrete part should be part of this document as they are dedicated to change.

[the remark this answers is not in the document]
Comment (kazydek) [reply docs/kyma/docs/005-overview-kyma.md]: Ok, I have only mentioned it without details.

File: docs/kyma/docs/005-overview-kyma.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/kyma/docs/005-overview-kyma.md]: As above.

**Pull request merged**